### PR TITLE
Many Extensions and Many Resources module

### PR DESCRIPTION
**Pull request by rsvoboda**
Many Extensions and Many Resources, module is using 47 Quarkus extension and 60 REST endpoints.
The intention is to ensure such kind of big application behaves well in various target environments. 

Went a bit beyond just back-porting 003-quarkus-many-extensions, enhanced the case a bit by generating REST resources and services, details in README of the module.

Fixes https://github.com/quarkus-qe/quarkus-test-suite/issues/76

### CODE REVIEW

**Review by Sgitario — COMMENTED** [review]
Just minor comments about mix of coverage between `http-minimum` and this new module.

File: BEEFY_TS_MIGRATION_STATUS.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 | Original Modules  | Second Header |
 | ------------- | ------------- |
 | 001-quarkus-getting-started-with-jaxrs | http/jaxrs |
+| 003-quarkus-many-extensions | super-size/many-extensions |
```
[the remark this answers is not in the document]
Comment (rsvoboda) [reply BEEFY_TS_MIGRATION_STATUS.md:5]: I see some future cases. Our customers do really interesting things, size of the apps is one of them.

I was also thinking that 201-large-static-content would be more appropriate fit here. I know it was planned for http module, but it's really Super Size Me scenario.

File: super-size/many-extensions/src/test/java/io/quarkus/ts/many/extensions/DevModeHttpMinimumIT.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.scenarios.annotations.DisabledOnNative;
+import io.quarkus.test.scenarios.annotations.DisabledOnQuarkusVersion;
+import io.quarkus.test.services.DevModeQuarkusApplication;
+import io.quarkus.test.utils.AwaitilityUtils;
+
+@QuarkusScenario
+@DisabledOnNative
+@DisabledOnQuarkusVersion(version = "1\\..*", reason = "Continuous Testing was entered in 2.x")
+public class DevModeHttpMinimumIT {
```
[the remark this answers is not in the document]
Comment (rsvoboda) [reply super-size/many-extensions/src/test/java/io/quarkus/ts/many/extensions/DevModeHttpMinimumIT.java:19]: Good catch / will do. I used http minimum as the starting point here.

[the remark this answers is not in the document]
Comment (rsvoboda) [reply super-size/many-extensions/src/test/java/io/quarkus/ts/many/extensions/DevModeHttpMinimumIT.java]: I have copied the tests, wanted to reuse them somehow, but `app.given()` is not defined in `BaseService` but in `RestService` and `DevModeQuarkusService`

[the remark this answers is not in the document]
Comment (rsvoboda) [reply super-size/many-extensions/src/test/java/io/quarkus/ts/many/extensions/DevModeHttpMinimumIT.java]: I think `.given()` should be defined in BaseService, I can see cases when one wants to ensure the same test coverage on both modes. What are your thoughts @Sgitario ?

[the remark this answers is not in the document]
Comment (Sgitario) [reply super-size/many-extensions/src/test/java/io/quarkus/ts/many/extensions/DevModeHttpMinimumIT.java]: Let's fix this for all: https://github.com/quarkus-qe/quarkus-test-framework/pull/160
Now, DevMode service will inherit from RestService (that contains the `.given()` method. 
So, you could have a base test class with your test cases and an abstract method to get the RestService app. 
A way to workaround this and have comment test cases now, it's to have an abstract method to get the `RequestSpecification given()`. This is what I did in https://github.com/quarkus-qe/quarkus-test-suite/pull/127/files#diff-a35f14131768db79aacc738139f5365b655de34efe8ef5a439a280de8b988985R19.

**Review by Sgitario — APPROVED** [review]

**Review by Sgitario — CHANGES_REQUESTED** [review]
native CI is failing

**Comment by rsvoboda** [comment]
New module requires xmx set to 5g, set as property for that module, testsuite wide default is 4g

File: super-size/many-extensions/src/test/java/io/quarkus/ts/many/extensions/DevModeHttpMinimumIT.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (rsvoboda) [reply super-size/many-extensions/src/test/java/io/quarkus/ts/many/extensions/DevModeHttpMinimumIT.java]: Cool, I can do that once new version of framework is out. Any plans / ETAs ?

**Comment by Sgitario** [comment]
run tests

**Comment by rsvoboda** [comment]
Failures in https://main-jenkins-csb-quarkusqe.apps.ocp4.prod.psi.redhat.com/job/quarkus-main-rhel8-jdk11-openshift-new-ts-jvm/48/ look like OpenShif env related and Jenkins job related (Non-resolvable parent POM for io.quarkus.ts.qe:many-extensions:1.0.0-SNAPSHOT: io.quarkus.ts.qe:parent:pom:1.0.0-SNAPSHOT was not found)

**Comment by Sgitario** [comment]
The OCP instances are again with network issues. 
I've also fixed a possible issue: https://github.com/quarkus-qe/quarkus-test-framework/commit/8e1175f1e544b188d166384ab0313d734cf98d7a which will be solved in the 0.0.4 release.
Pls, rebase your PR and I'm fine to merge this.

**Review by Sgitario — APPROVED** [review]

**Comment by rsvoboda** [comment]
JVM run failure is not related to this PR.

Failure in  `Security: Keycloak + OIDC Client + Extended`

```
2021-08-03T07:03:34.6943556Z [ERROR] Failures: 
2021-08-03T07:03:34.6945720Z [ERROR]   OpenApiIT.testJsonOpenApiPathAccessResource:46->assertContent:76 
2021-08-03T07:03:34.6947304Z Expected: is "{\"title\":\"Generated API\",\"version\":\"1.0\"}"
2021-08-03T07:03:34.6949212Z      but: was "{\"title\":\"security-keycloak-oidc-client-extended API\",\"version\":\"1.0.0-SNAPSHOT\"}"
2021-08-03T07:03:34.6951504Z [ERROR]   OpenApiIT.testYamlOpenApiPathAccessResource:53->assertContent:76 
2021-08-03T07:03:34.6953161Z Expected: is "{\"title\":\"Generated API\",\"version\":\"1.0\"}"
2021-08-03T07:03:34.6954737Z      but: was "{\"title\":\"security-keycloak-oidc-client-extended API\",\"version\":\"1.0.0-SNAPSHOT\"}"
```

**Comment by rsvoboda** [comment]
@Sgitario I think this can be merged now.

**Pull request merged**